src/averageAges.js

```diff
@@ -54,8 +62,24 @@ function calculateWomenAverageAge(people, withChildren) {
  *
  * @return {number}
  */
-function calculateAverageAgeDiff(people, onlyWithSon) {
-  // write code here
+function calculateAverageAgeDiff(people, onlyWithSon = false) {
+  const filteredPeople = onlyWithSon
+    ? people
+      .filter(person => person.sex === 'm' && people
+        .find(mother => mother.name === person.mother))
+    : people
+      .filter(person => people
+        .find(mother => mother.name === person.mother));
```

Comment:
Seems that you repeat the same find method twice. Try to refactor it to avoid repetition